autogluon/utils/tabular/ml/models/lgb/lgb_model.py

```diff
@@ -125,26 +127,16 @@ def fit(self, X_train=None, Y_train=None, X_test=None, Y_test=None, dataset_trai
     def predict_proba(self, X, preprocess=True):
         if preprocess:
             X = self.preprocess(X)
-        if self.problem_type == REGRESSION:
-            return self.model.predict(X)
 
         y_pred_proba = self.model.predict(X)
-        if self.problem_type == BINARY:
-            if len(y_pred_proba.shape) == 1:
-                return y_pred_proba
-            elif y_pred_proba.shape[1] > 1:
-                return y_pred_proba[:, 1]
-            else:
-                return y_pred_proba
-        elif self.problem_type == MULTICLASS:
-            return y_pred_proba
-        else:
-            if len(y_pred_proba.shape) == 1:
-                return y_pred_proba
-            elif y_pred_proba.shape[1] > 2:  # Should this ever happen?
-                return y_pred_proba
-            else:  # Should this ever happen?
-                return y_pred_proba[:, 1]
+        if (self.problem_type == BINARY) and (len(y_pred_proba.shape) > 1) and (y_pred_proba.shape[1] > 1):
```

Comment:
The previous code was quite verbose, but it was that way because it minimized the amount of checks required of the data, and thus I think is faster than the condensed version and recommend this change be reverted. Instead of return y_pred_proba, just set y_pred_proba=, then have the normalization code at the end as currently done.

Reply:
done